src/Application/Commands/ExpirePendingMandates.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Application\Commands;
+
+use MatchBot\Domain\DomainException\CouldNotCancelStripePaymentIntent;
+use MatchBot\Domain\DonationRepository;
+use MatchBot\Domain\DonationService;
+use MatchBot\Domain\MandateCancellationType;
+use MatchBot\Domain\RegularGivingMandateRepository;
+use MatchBot\Domain\RegularGivingService;
+use Psr\Clock\ClockInterface;
+use Psr\Log\LoggerInterface;
+use Ramsey\Uuid\Uuid;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+#[AsCommand(
+    name: 'matchbot:expire-pending-mandates',
+    description: "Cancels regular giving mandates left pending (e.g. while donor considered 3DS authentication)",
+)]
+class ExpirePendingMandates extends LockingCommand
+{
+    public function __construct(
+        private RegularGivingMandateRepository $regularGivingMandateRepository,
+        private RegularGivingService $regularGivingService,
+        private ClockInterface $clock,
+        private LoggerInterface $log,
+    ) {
+        parent::__construct();
+    }
+
+    protected function doExecute(InputInterface $input, OutputInterface $output): int
+    {
+        $fiveMinutes = new \DateInterval('PT5M');
```

Comment:
This seems a little low to me. I know at least one bank I use has a 9 minute timeout on their side; might be worth a straw poll of the team to get more examples. My instinct is to go for around 15 minutes, to minimise the risk that 3DS could succeed and then we still have an expired mandate. (I'm not yet clear what happens if a donor hits this scenario?)

Reply:
We cancel the payment attempt via the stripe API when this expires, so the donor should get a payment failure message and our DB will be left basically as it would be if they had just walked away when they saw the 3DS popup. I think I asked most of the team yesterday, 5 seemed OK but I can't immediately think what the risk would be if its too high, although it does become more important to proactively cancel if they fail authentication or if they close the tab and then attempt to start a new mandate.